lib/HttpClient.ts

```diff
@@ -134,6 +135,8 @@ export class HttpClient {
             if (requestOptions.maxRedirects != null) {
                 this._maxRedirects = Math.max(requestOptions.maxRedirects, 0);
             }
+
+            this._usePrivateAgent = !!this.requestOptions.usePrivateAgent;
```

Comment:
at first I thought, well let's just always re-use the agent for perf.  When wouldn't you want to.  The argument was brought that some calls are to proxy, some not etc...  But, constructor of this http client takes proxy config.  So if you wanted to make some calls to proxy, some not, you already need to contruct two instances.

However, I do acknowledge that there might be a case where you don't want to re-use within a single instance of the http client.  I can't come up with it but it might exist. 

That's what brought me to the conclusion that we should still have this option but it should be an option to *not* reuse agent. 

Other option I'm toying with is whether we have a proxy config with an exclusion list but maybe that's a higher level concept?  I need to look at .net http clients for patterns.  Will follow up.

Reply:
I think we should follow the .net pattern of taking a bypass list of urls for proxy config.

https://msdn.microsoft.com/en-us/library/ms144293(v=vs.110).aspx

If we do that, then there is one proxy config.  That means the request code here should reuse an agent for non proxy calls and potentially one more if a proxy call.  

So, I think this should morph into "supporting proxy exclusion lists" and the agent should pass through since it supports exclusion lists.

Reply:
That seems fine to me, I have updated the PR to have two agents one for proxy and another one for non-proxy and use the correct one based on the bypass list.